### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by royclarkson**
- Only log potentially sensitive information in debug mode
- Apply consistent style

### CODE REVIEW

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java [hunk 1]
```diff
@@ -108,14 +121,24 @@ public AppDeploymentCreateServiceInstanceWorkflow(BrokeredServices brokeredServi
...
+					request.getServiceDefinition().getName(), request.getPlan().getName());
+				LOG.debug(REQUEST_LOG_TEMPLATE, request);
+			})
+			.doOnComplete(() -> {
+				LOG.info("Finish deploying backing applications. serviceDefinitionName={}, planName={}",
+					request.getServiceDefinition().getName(), request.getPlan().getName());
+				LOG.debug(REQUEST_LOG_TEMPLATE, request);
+			})
+			.doOnError(e -> {
+				if (LOG.isErrorEnabled()) {
```
Comment (spikymonkey) [inline spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java:135]: There are a few checks like this in the PR and I'm wondering if they're worth it. For debug, info levels, sure... but does anyone ever disable error level logging?

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java [hunk 2]
```diff
@@ -138,14 +153,24 @@ public AppDeploymentDeleteServiceInstanceWorkflow(BrokeredServices brokeredServi
...
-				exception -> log.error(String.format("Error undeploying backing applications for %s/%s with error '%s'",
-					request.getServiceDefinition().getName(), request.getPlan().getName(), exception.getMessage()),
-					exception));
+			.doOnRequest(l -> {
+				LOG.info("Undeploying backing applications. serviceDefinitionName={}, planName={}",
+					request.getServiceDefinition().getName(), request.getPlan().getName());
+				LOG.debug(REQUEST_LOG_TEMPLATE, request);
+			})
+			.doOnComplete(() -> {
+				LOG.debug("Finish undeploying backing applications. serviceDefinitionName={}, planName={}",
```
Comment (spikymonkey) [inline spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java:162]: Think this one should be `.info`?

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java [hunk 3]
```diff
@@ -547,10 +606,13 @@ private String getLastUpdatedPackageId(List<PackageResource> packageResources) {
...
-			.doOnRequest(l -> LOG.debug("Creating package for application {}", applicationId))
-			.doOnSuccess(response -> LOG.info("Created package for application {}", applicationId))
-			.doOnError(e -> LOG
-				.warn(String.format("Error creating package for application %s: %s", applicationId, e.getMessage())));
+			.doOnRequest(l -> LOG.debug("Creating package. applicationId={}", applicationId))
+			.doOnSuccess(response -> {
+				LOG.info("Success creating package. applicationId={}", applicationId);
+				LOG.debug(RESPONSE_LOG_TEMPLATE, response);
+			})
+			.doOnError(e -> LOG.warn(String.format("Error creating package. applicationId=%s, " + ERROR_LOG_TEMPLATE,
```
Comment (spikymonkey) [inline spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:614]: Should this be logged at error level?

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java [hunk 4]
```diff
@@ -565,23 +627,25 @@ private String getLastUpdatedPackageId(List<PackageResource> packageResources) {
...
 
-	private Mono<Void> pushApplication(DeployApplicationRequest request,
-		Map<String, String> deploymentProperties,
+	private Mono<Void> pushApplication(DeployApplicationRequest request, Map<String, String> deploymentProperties,
 		Resource appResource) {
 		ApplicationManifest manifest = buildAppManifest(request, deploymentProperties, appResource);
 
 		if (LOG.isDebugEnabled()) {
-			LOG.debug("Pushing manifest" + manifest.toString());
+			LOG.debug("Pushing app manifest. manifest={}" + manifest.toString());
```
Comment (spikymonkey) [inline spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:648]: We've got a format string combined with string concatenation here 😄  It'll log an extra pair of `{}` - `... manifest={}<manifest contents>`

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java [hunk 5]
```diff
@@ -747,27 +817,30 @@ else if (StringUtils.hasText(targetProperties.getUsername())) {
...
+				spaceName, e.getMessage()), e))
 			.flatMap(spaceId -> this.client.spaces()
 				.delete(DeleteSpaceRequest.builder()
 					.spaceId(spaceId)
 					.recursive(true)
 					.build())
 				.then())
-			.doOnError(exception -> LOG.debug("Error deleting space {} with error '{}'",
-				spaceName, exception.getMessage()))
+			.doOnError(e -> LOG.debug(String.format("Error deleting space. spaceName=%s, " + ERROR_LOG_TEMPLATE,
```
Comment (spikymonkey) [inline spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:842]: Should this be logged at error level?

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java [hunk 6]
```diff
@@ -1124,13 +1199,12 @@ private Path getApplication(Resource resource) {
...
 		return cloudFoundryOperations.services().deleteInstance(
-			org.cloudfoundry.operations.services.DeleteServiceInstanceRequest
-				.builder()
+			org.cloudfoundry.operations.services.DeleteServiceInstanceRequest.builder()
 				.name(serviceInstanceName)
 				.completionTimeout(apiPollingTimeout(deploymentProperties))
 				.build())
-			.doOnError(exception -> LOG.debug(String.format("Error deleting service instance %s with error '%s'",
-				serviceInstanceName, exception.getMessage()), exception))
+			.doOnError(e -> LOG.debug(String.format("Error deleting service instance. serviceInstanceName=%s, " +
```
Comment (spikymonkey) [inline spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:1206]: Log at error level?

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/service/WorkflowServiceInstanceBindingService.java [hunk 7]
```diff
@@ -218,10 +224,16 @@ else if (isRouteBindingRequest(request)) {
...
-				.doOnRequest(l -> LOG.debug("Deleting service instance binding"))
-				.doOnComplete(() -> LOG.debug("Finished deleting service instance binding"))
-				.doOnError(exception -> LOG.error(String.format("Error deleting service instance binding with error " +
-					"'%s'", exception.getMessage()), exception)))
+				.doOnRequest(l -> {
+					LOG.info("Deleting service instance binding");
+					LOG.debug("request={}", request);
+				})
+				.doOnComplete(() -> {
+					LOG.debug("Finish deleting service instance binding");
```
[the remark this answers is not in the document]
Comment (royclarkson) [reply spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/service/WorkflowServiceInstanceBindingService.java:232]: Yep 👍  copy/paste strikes again.

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java (hunk 1 above)
Comment (royclarkson) [reply spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java:135]: pmd complains about not having those in some cases. We could make a choice to disable that check or live with it? Apparently some of the logging APIs have those checks built in, which is why they aren't everywhere.

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java (hunk 2 above)
Comment (royclarkson) [reply spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentDeleteServiceInstanceWorkflow.java:162]: yep 👍

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java (hunk 1 above)
Comment (spikymonkey) [reply spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java:135]: Oh, interesting - I had no idea. I don't feel strongly on this at all, it just struck me as odd that we were doing it in particular places but not others. Happy to leave as it is.

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java (hunk 3 above)
Comment (royclarkson) [reply spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:614]: When I was reviewing `CloudFoundryAppDeployer`, I didn't change any of the existing levels. There are many warns in there, where we use error everywhere else. I like standardizing on error, so I'm glad you caught that.

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java (hunk 4 above)
Comment (royclarkson) [reply spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java:648]: oops

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java (hunk 1 above)
Comment (royclarkson) [reply spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/workflow/instance/AppDeploymentCreateServiceInstanceWorkflow.java:135]: I removed a couple of these checks, but others pmd still wanted. 🤷‍♂️

File: spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java (hunk 3 above)
Comment (royclarkson) [reply spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java]: Changed all of the warns to errors

**Pull request merged**